app/controllers/games_controller.rb

```diff
@@ -25,14 +25,24 @@ def update
 		redirect_to game_path(@game)
 	end
 
-	def new
+  def concede
+    @game = Game.find(params[:id])
+    @black_player = User.find_by(id: @game.black_uid)
+    @white_player = User.find_by(id: @game.white_uid)
+    @current_player = current_user.id
+      if @black_player == @current_player
+        @game.winner_id = @white_player
+      else
+        @game.winner_id = @black_player
```

Comment:
again black player is an object.  You don't want to assign and object to the integer.

this should look like @game.winner_id = @black_player.id